src/screens/Settings/index.js

```diff
@@ -0,0 +1,151 @@
+/* @flow */
+import React, { Component } from "react";
+import {
+  ScrollView,
+  View,
+  Text,
+  StyleSheet,
+  Alert,
+  Switch,
+  Image
+} from "react-native";
+import { connect } from "react-redux";
+import LText from "../../components/LText";
+import { withReboot } from "../../components/RebootContext";
+import SectionEntry from "../../components/SectionEntry";
+import SectionTitle from "../../components/SectionTitle";
+import { getLabelFromRange } from "./ChartTimeRange";
+import { saveSettings } from "../../actions/settings";
+import type { State } from "../../reducers";
+
+const mapStateToProps = (state: State) => ({
+  settings: state.settings
+});
+
+const mapDispatchToProps = {
+  saveSettings
+};
+
+class SignOut_ extends Component<{ reboot: (?boolean) => * }> {
+  onResetAll = async () => {
+    await this.props.reboot(true);
+  };
+  onSignOut = () => {
+    Alert.alert(
+      "Are you sure you want to sign out?",
+      "All accounts data will be removed from your phone.",
+      [
+        { text: "Cancel", style: "cancel" },
+        { text: "Sign me out", onPress: this.onResetAll }
+      ]
+    );
+  };
+  render() {
+    return (
+      <View style={{ marginVertical: 40 }}>
+        <SectionEntry onPress={this.onSignOut} center>
+          <Text style={styles.signOutText}>Sign Out</Text>
+        </SectionEntry>
+      </View>
+    );
+  }
+}
+
+const SignOut = withReboot(SignOut_);
```

Comment:
we might want to split a bit the component. like SignOut can be in its own file. i know you didn't modified this, so if you want, we can do it later.